spec/validators/collection_membership_validator_spec.rb

```diff
@@ -0,0 +1,212 @@
+        end
+      end
+
+      context 'and work is added to collections' do
+        let(:col1) { FactoryBot.valkyrie_create(:hyrax_collection) }
+        let(:col2) { FactoryBot.valkyrie_create(:hyrax_collection) }
+        let(:mem_of_cols_attrs) do
+          { "0" => { "id" => col1.id.to_s, "_destroy" => "false" },
+            "1" => { "id" => col2.id.to_s, "_destroy" => "false" } }
+        end
+
+        it 'validates and sets member_of_collection_ids to new collections' do
+          validator.validate(form)
+
+          expect(form.errors).to be_blank
+          expect(form.member_of_collection_ids).to contain_exactly(col1.id, col2.id)
+        end
+
+        context 'when it is already in a collection' do
+          let(:work) { FactoryBot.build(:hyrax_work, :as_collection_member) }
+          let(:col_id) { work.member_of_collection_ids.first.id }
+
+          it 'validates and replaces member_of_collection_ids with new collection set' do
+            validator.validate(form)
+
+            expect(form.errors).to be_blank
+            expect(form.member_of_collection_ids).to contain_exactly(col1.id, col2.id)
+          end
+        end
+
+        context 'and collection type does not allow multiple membership' do
+          let(:single_mem_col_type) { FactoryBot.create(:collection_type, allow_multiple_membership: false) }
+
+          context 'and work is in another collection NOT posing a conflict' do
+            let(:work) { FactoryBot.build(:hyrax_work, :as_collection_member) }
+            let(:col_id) { work.member_of_collection_ids.first.id }
+            let(:sm_col) { FactoryBot.valkyrie_create(:hyrax_collection, collection_type_gid: single_mem_col_type.to_global_id) }
+            let(:mem_of_cols_attrs) do
+              { "0" => { "id" => col_id.to_s, "_destroy" => "false" },
+                "1" => { "id" => sm_col.id.to_s, "_destroy" => "false" } }
+            end
+
+            it 'validates and appends new collections to member_of_collection_ids' do
+              validator.validate(form)
+
+              expect(form.errors).to be_blank
+              expect(form.member_of_collection_ids).to contain_exactly(col_id, sm_col.id)
+            end
+          end
+
+          context 'and work is in another collection that IS posing a conflict' do
+            let(:work) { FactoryBot.build(:hyrax_work, member_of_collection_ids: [sm_col1.id]) }
+            let(:sm_col1) { FactoryBot.valkyrie_create(:hyrax_collection, collection_type_gid: single_mem_col_type.to_global_id) }
+            let(:sm_col2) { FactoryBot.valkyrie_create(:hyrax_collection, collection_type_gid: single_mem_col_type.to_global_id) }
+            let(:mem_of_cols_attrs) do
+              { "0" => { "id" => sm_col1.id.to_s, "_destroy" => "false" },
+                "1" => { "id" => sm_col2.id.to_s, "_destroy" => "false" } }
+            end
+
+            it 'fails validating and sets errors' do
+              validator.validate(form)
+
```

Comment:
Is it worth testing here (or at least documenting in tests) how the form's member_of_collection_ids is set?
```suggestion
expect(form.member_of_collection_ids).to contain_exactly(sm_col1.id, sm_col2.id)
```